bin/kafka-run-class.sh

```diff
@@ -243,6 +243,14 @@ fi
 (( WINDOWS_OS_FORMAT )) && LOG_DIR=$(cygpath --path --mixed "${LOG_DIR}")
 KAFKA_LOG4J_CMD_OPTS="-Dkafka.logs.dir=$LOG_DIR $KAFKA_LOG4J_OPTS"
 
+# slf4j provider settings
+if [ -z "$SLF4J_PROVIDER" ]; then
+  SLF4J_PROVIDER="org.slf4j.reload4j.Reload4jServiceProvider"
+fi
+
+# Add the slf4j provider to KAFKA_LOG4J_CMD_OPTS
+KAFKA_LOG4J_CMD_OPTS="$KAFKA_LOG4J_CMD_OPTS -Dslf4j.provider=$SLF4J_PROVIDER"
```

Comment:
This is a new system property, so KAFKA_LOG4J_CMD_OPTS shouldn't already contain -Dslf4j.provider
👍